File: backend/internal/types/template.go
package types

import "time"

type TemplateDTO struct {
	ID           uint64    `json:"id"`
	UserID       uint64    `json:"user_id"`
	Name         string    `json:"name"`
	Subject      string    `json:"subject"`
	HTMLContent  string    `json:"html_content"`
	TextContent  string    `json:"text_content"`
	ThumbnailURL string    `json:"thumbnail_url"`
	IsDefault    bool      `json:"is_default"`
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}

type CreateTemplateRequest struct {
	UserID      uint64 `json:"-"`
	Name        string `json:"name" binding:"required"`
	Subject     string `json:"subject" binding:"required"`
	HTMLContent string `json:"html_content" binding:"required"`
	TextContent string `json:"text_content"`
	IsDefault   bool   `json:"is_default"`
}

type UpdateTemplateRequest struct {
	UserID      uint64
	Name        string `json:"name"`
	Subject     string `json:"subject"`
	HTMLContent string `json:"html_content"`
	TextContent string `json:"text_content"`
	IsDefault   *bool  `json:"is_default"`
}

type PreviewTemplateRequest struct {
	HTMLContent string                 `json:"html_content" binding:"required"`
	Variables   map[string]interface{} `json:"variables"`
}

type UploadTemplateImageResponse struct {
	URL string `json:"url"`
}

File: frontend/types/template.ts
export interface Template {
    id: number;
    user_id: number;
    name: string;
    subject: string;
    type: 'mjml' | 'html';
    html_content: string;
    mjml_content?: string;
    text_content?: string;
    thumbnail_url?: string;
    is_default: boolean;
    created_at: string;
    updated_at: string;
}

export interface CreateTemplateRequest {
    name: string;
    subject: string;
    type: 'mjml' | 'html';
    html_content: string;
    mjml_content?: string;
    text_content?: string;
    is_default?: boolean;
}

export interface UpdateTemplateRequest {
    name?: string;
    subject?: string;
    type?: 'mjml' | 'html';
    html_content?: string;
    mjml_content?: string;
    text_content?: string;
    is_default?: boolean;
}

export interface PreviewTemplateRequest {
    html_content: string;
    variables?: Record<string, any>;
}

export interface TemplateListResponse {
    data: Template[];
    page: number;
    limit: number;
    total: number;
}
